cmd/webauthn-oidc-idp/keyset.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/lstoll/tinkrotate"
+	"github.com/oklog/run"
+	"github.com/tink-crypto/tink-go/v2/aead"
+	"github.com/tink-crypto/tink-go/v2/jwt"
+	"github.com/tink-crypto/tink-go/v2/keyset"
+	"github.com/tink-crypto/tink-go/v2/proto/tink_go_proto"
+)
+
+type Keyset struct {
+	// Name of the keyset, used to refer to it in the store
+	Name string
+	// Template for new keys in this set
+	Template *tink_go_proto.KeyTemplate
+	// RotateEvery indicates how often we should rotate a new key in.
+	RotateEvery time.Duration
+}
+
+const (
+	keysetIDCookie = "cookie"
+	keysetIDOIDC   = "oidc"
+)
+
+var (
+	cookieRotatePolicy = tinkrotate.RotationPolicy{
+		KeyTemplate:         aead.XAES256GCM192BitNonceKeyTemplate(),
+		PrimaryDuration:     30 * 24 * time.Hour,
+		PropagationTime:     24 * time.Hour,
+		PhaseOutDuration:    7 * 24 * time.Hour,
+		DeletionGracePeriod: 0,
+	}
+	oidcRotatePolicy = tinkrotate.RotationPolicy{
+		KeyTemplate:         jwt.RS256_2048_F4_Key_Template(),
+		PrimaryDuration:     24 * time.Hour,
+		PropagationTime:     6 * time.Hour,
+		PhaseOutDuration:    24 * time.Hour,
+		DeletionGracePeriod: 0,
+	}
+)
+
+func initKeysets(ctx context.Context, db *sql.DB, g run.Group) (cookieKeyset, oidcKeyset *KeysetHandles, _ error) {
```

Comment:
This is noisier than I'd like, opened https://github.com/lstoll/tinkrotate/issues/9